src/components/v2/V2Project/V2FundingCycleSection/FundingCycleHistory.tsx

```diff
@@ -0,0 +1,281 @@
+  numFundingCycles,
+  index,
+  setSelectedIndex,
+}: {
+  fundingCycle: V2FundingCycle
+  numFundingCycles: number
+  index: number
+  setSelectedIndex: (index: number) => void
+}) {
+  const { projectId, primaryTerminal } = useContext(V2ProjectContext)
+
+  const {
+    theme: { colors },
+  } = useContext(ThemeContext)
+
+  const { data: distributionLimitData } = useProjectDistributionLimit({
+    projectId,
+    configuration: fundingCycle?.configuration?.toString(),
+    terminal: primaryTerminal,
+  })
+
+  const { data: usedDistributionLimit } = useUsedDistributionLimit({
+    projectId,
+    terminal: primaryTerminal,
+    fundingCycleNumber: fundingCycle?.number,
+  })
+
+  const [distributionLimit, distributionLimitCurrency] =
+    distributionLimitData ?? []
+
+  const distributionLimitIsInfinite = distributionLimit?.eq(
+    MAX_DISTRIBUTION_LIMIT,
+  )
+  const distributionLimitIsZero = !distributionLimit || distributionLimit?.eq(0)
+
+  return (
+    <div
+      key={fundingCycle.number.toString()}
+      onClick={() => setSelectedIndex(index)}
+      style={{
+        display: 'flex',
+        alignItems: 'baseline',
+        justifyContent: 'space-between',
+        cursor: 'pointer',
+        ...(index < numFundingCycles - 1
+          ? {
+              paddingBottom: 20,
+              borderBottom: '1px solid ' + colors.stroke.tertiary,
+            }
+          : {}),
+      }}
+    >
+      <Space align="baseline">
+        <h3>#{fundingCycle.number.toString()}</h3>
+
+        <div style={{ fontSize: '.8rem', marginLeft: 10 }}>
+          <CurrencySymbol
+            currency={V2CurrencyName(
+              distributionLimitCurrency?.toNumber() as V2CurrencyOption,
+            )}
+          />
+          {!(distributionLimitIsInfinite || distributionLimitIsZero) ? (
+            <>
+              <Trans>
+                {formatWad(usedDistributionLimit, { precision: 2 })}/
+                {formatWad(distributionLimit, { precision: 2 })} withdrawn
+              </Trans>
+            </>
+          ) : (
+            <>
+              <Trans>
+                {formatWad(usedDistributionLimit, { precision: 2 })} withdrawn
+              </Trans>
+            </>
+          )}
+        </div>
+      </Space>
+
+      <div style={{ flex: 1 }}></div>
+
+      <Space align="baseline" style={{ fontSize: '.8rem' }}>
+        {formatHistoricalDate(
+          fundingCycle.start.add(fundingCycle.duration).mul(1000).toNumber(),
+        )}
+        <CaretRightOutlined />
+      </Space>
+    </div>
+  )
+}
+
+export default function FundingCycleHistory() {
+  const { projectId, fundingCycle: currentFundingCycle } =
+    useContext(V2ProjectContext)
+  const { contracts } = useContext(V2UserContext)
+
+  const [selectedIndex, setSelectedIndex] = useState<number>()
+  const [pastFundingCycles, setPastFundingCycles] = useState<V2FundingCycle[]>(
+    [],
+  )
+
+  useEffect(() => {
+    const loadPastFundingCycles = async () => {
```

Comment:
> **suggestion** this function is quite large and hard to follow. WDYT about isolating the logic into smaller functions to make it more readable/testable?

TBH I don't think it would help very much.